common/domain/handler_test.go

```diff
@@ -244,3 +342,233 @@ func TestRegisterDomain(t *testing.T) {
+
+	domainName := "test-domain"
+	domainID := "test-domain-id"
+	isGlobalDomain := true
+	failoverVersion := int64(123)
+	lastUpdatedTime := time.Now().UnixNano()
+	failoverEndTime := lastUpdatedTime + int64(time.Hour)
+	var configIsolationGroup types.IsolationGroupConfiguration
+
+	domainInfo := &persistence.DomainInfo{
+		ID:          domainID,
+		Name:        domainName,
+		Status:      persistence.DomainStatusRegistered,
+		Description: "Test Domain",
+		OwnerEmail:  "test@uber.com",
+		Data:        map[string]string{"k1": "v1"},
+	}
+	domainConfig := &persistence.DomainConfig{
+		Retention: 24,
+	}
+	domainReplicationConfig := &persistence.DomainReplicationConfig{
+		ActiveClusterName: "active",
+		Clusters:          []*persistence.ClusterReplicationConfig{{ClusterName: "active"}, {ClusterName: "standby"}},
+	}
+
+	mockDomainManager.EXPECT().
+		GetDomain(gomock.Any(), &persistence.GetDomainRequest{Name: domainName}).
+		Return(&persistence.GetDomainResponse{
+			Info:              domainInfo,
+			Config:            domainConfig,
+			ReplicationConfig: domainReplicationConfig,
+			IsGlobalDomain:    isGlobalDomain,
+			FailoverVersion:   failoverVersion,
+			FailoverEndTime:   &failoverEndTime,
+			LastUpdatedTime:   lastUpdatedTime,
+		}, nil)
+
+	describeRequest := &types.DescribeDomainRequest{Name: &domainName}
+	expectedResponse := &types.DescribeDomainResponse{
+		IsGlobalDomain:  isGlobalDomain,
+		FailoverVersion: failoverVersion,
+		FailoverInfo: &types.FailoverInfo{
+			FailoverVersion:         failoverVersion,
+			FailoverStartTimestamp:  lastUpdatedTime,
+			FailoverExpireTimestamp: failoverEndTime,
+		},
+		DomainInfo: &types.DomainInfo{
+			Name:        domainName,
+			Status:      types.DomainStatusRegistered.Ptr(),
+			Description: "Test Domain",
+			OwnerEmail:  "test@uber.com",
+			Data:        map[string]string{"k1": "v1"},
+			UUID:        domainID,
+		},
+		Configuration: &types.DomainConfiguration{
+			WorkflowExecutionRetentionPeriodInDays: 24,
+			EmitMetric:                             false,
+			BadBinaries:                            &types.BadBinaries{Binaries: nil},
+			HistoryArchivalStatus:                  types.ArchivalStatusDisabled.Ptr(),
+			HistoryArchivalURI:                     "",
+			VisibilityArchivalStatus:               types.ArchivalStatusDisabled.Ptr(),
+			VisibilityArchivalURI:                  "",
+			IsolationGroups:                        &configIsolationGroup,
+			AsyncWorkflowConfig:                    &types.AsyncWorkflowConfiguration{},
+		},
+		ReplicationConfiguration: &types.DomainReplicationConfiguration{
+			ActiveClusterName: "active",
+			Clusters: []*types.ClusterReplicationConfiguration{
+				{ClusterName: "active"},
+				{ClusterName: "standby"},
+			},
+		},
+	}
+
+	resp, err := handler.DescribeDomain(context.Background(), describeRequest)
+	assert.NoError(t, err)
+	assert.EqualValues(t, expectedResponse, resp)
+
+	mockDomainManager.EXPECT().
```

Comment:
nit: my understanding is that this is testing a different scenario. Maybe organize the test with have test names so it's easier to understand

Reply:
It is just a err!= nil  1 case. that is why did add table test etc. But you are right, i ll add the comment whuich case it demonstrates in next diff.